From 20e5ec488a12121670f97d754e1a21c0ebeda285 Mon Sep 17 00:00:00 2001
From: Takashi Yano <takashi.yano@nifty.ne.jp>
Date: Wed, 30 Mar 2022 12:46:08 +0900
Subject: [PATCH 056/N] Cygwin: pty: Fix crash on master close in Windows 7.

- The 4th parameter of WriteFile() cannot be NULL especially in
  Windows 7 as mentioned in Microsoft documentation. This patch
  fixes that.

Addresses: https://cygwin.com/pipermail/cygwin/2022-March/251162.html
Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>
---
 winsup/cygwin/fhandler_tty.cc | 2 +-
 winsup/cygwin/release/3.3.5   | 3 +++
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/winsup/cygwin/fhandler_tty.cc b/winsup/cygwin/fhandler_tty.cc
index f78dc76..3675250 100644
--- a/winsup/cygwin/fhandler_tty.cc
+++ b/winsup/cygwin/fhandler_tty.cc
@@ -2172,7 +2172,7 @@ fhandler_pty_master::close ()
 	    }
 	  release_output_mutex ();
 	  get_ttyp ()->stop_fwd_thread = true;
-	  WriteFile (to_master_nat, "", 0, NULL, NULL);
+	  WriteFile (to_master_nat, "", 0, &len, NULL);
 	  master_fwd_thread->detach ();
 	}
     }
diff --git a/winsup/cygwin/release/3.3.5 b/winsup/cygwin/release/3.3.5
index d2a7f77..9d44c1b 100644
--- a/winsup/cygwin/release/3.3.5
+++ b/winsup/cygwin/release/3.3.5
@@ -43,3 +43,6 @@ Bug Fixes
 
 - Fix a formatting problem in gmondump where all displayed addresses are
   mistakenly prefixed with "0x0x".
+
+- Fix crash on pty master close in Windows 7.
+  Addresses: https://cygwin.com/pipermail/cygwin/2022-March/251162.html
-- 
2.9.0

